core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -0,0 +1,423 @@
+package org.bitcoins.core.psbt
+
+import org.bitcoins.core.crypto.{ECDigitalSignature, ECPublicKey, ExtPublicKey}
+import org.bitcoins.core.hd.BIP32Path
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.{CompactSizeUInt, NetworkElement}
+import org.bitcoins.core.protocol.script.{
+  EmptyScriptSignature,
+  ScriptPubKey,
+  ScriptSignature,
+  ScriptWitness
+}
+import org.bitcoins.core.protocol.transaction.{
+  BaseTransaction,
+  EmptyWitness,
+  Transaction,
+  TransactionOutput,
+  WitnessTransaction
+}
+import org.bitcoins.core.psbt.GlobalPSBTRecord.UnsignedTransaction
+import org.bitcoins.core.psbt.PSBTGlobalKey.UnsignedTransactionKey
+import org.bitcoins.core.script.crypto.HashType
+import org.bitcoins.core.util.Factory
+import scodec.bits._
+
+case class PSBT(
+    globalMap: GlobalPSBTMap,
+    inputMaps: Vector[InputPSBTMap],
+    outputMaps: Vector[OutputPSBTMap],
+    tx: Transaction)
```

Comment:
I think that rather than having `tx` as a param, you should make it a `val` in side the definition of `PSBT` by pulling it out of `globalMap` and you also should have a `require` statement that the `globalMap` contains a tx as it says in the BIP that `A PSBT must have a transaction, otherwise it is invalid.`